src/Lachain.Networking/Hub/HubConnector.cs

```diff
@@ -100,13 +106,53 @@ public void Send(byte[] publicKey, byte[] message)
             CommunicationHub.Net.Hub.Send(publicKey, CompressUtils.DeflateCompress(message).ToArray());
         }
 
+        public void TrySend(byte[] publicKey, byte[] message)
+        {
+            lock (_messageQueue)
+            {
+                _messageQueue.Enqueue((publicKey, message));
+                Monitor.PulseAll(_messageQueue);
+            }
+        }
+
+        private void SendMessages()
+        {
+            const int waitForSomeTime = 500;
+            while (_running)
+            {
+                try
+                {               
+                    lock (_messageQueue)
+                    {
+                        while (_messageQueue.Count == 0)
+                            Monitor.Wait(_messageQueue);
+
+                        while (_messageQueue.Count > 0)
+                        {
+                            byte[] publicKey;
+                            byte[] message;
+                            (publicKey, message) = _messageQueue.Dequeue();
+                            Send(publicKey, message);
+                        }
+                    }       
+
+                    Thread.Sleep(waitForSomeTime);
```

Comment:
Why we need thios sleep here after waiting monitor at the cycle beginnig?

Reply:
I am really not sure about the design here. All clientworker use this to send non priority message and lock this so I added sleep. priority messages go directly to Send method. In case of spam request (like block request) or pool sync (we have pool sync with one tx at a time, maybe  we can send several txes together?) this method can be locked for long time. So I added sleep. But I am open to suggestions.

Reply:
I have removed this way of sending message for now, will test and see what happpes.